VideoLocker/src/main/java/org/edx/mobile/util/LocaleUtils.java

```diff
@@ -0,0 +1,42 @@
+package org.edx.mobile.util;
+
+import android.support.annotation.NonNull;
+
+import java.util.Locale;
+
+public class LocaleUtils {
+
+    public static String getCountryNameFromCode(@NonNull String countryCode) throws InvalidLocaleException {
+        final String uppercaseRegion = countryCode.toUpperCase(Locale.ROOT);
+        if (!isValidBcp47Alpha(uppercaseRegion, 2, 2)) {
```

Comment:
@1zaman That check is guaranteed to be false, since it requires a length of 3, while we already check that the length is exactly 2.